### DISCUSSION TITLE
Regression: URL protocol removed

### DISCUSSION THREAD

**Discussion by WorkInProgress-Development** [will-stone/browserosaurus#527] [General]
The URL protocol (e.g. https://) and www. are removed when received by Browserosaurus.
As a result, the following issues can occur:
1) Subdomains beginning with www (e.g. the still sometimes used www2) do not show in the preview, but rather have the www and the next character removed. (This is a bug which maybe should be an issue)

url used here: http://www2.example.org
<img width="362" src="https://user-images.githubusercontent.com/38821590/165001370-4dd8ad19-4610-45ac-a31a-883563bb9b20.png">

url used here: https://wwwtest.example.org
<img width="362" alt="Screen Shot 2022-04-25 at 9 35 28 am" src="https://user-images.githubusercontent.com/38821590/165001317-65ed2d47-e912-4857-a80b-1a3d6b42481f.png">

2) The protocol and subdomain are not visible when previewing, so a user would need to copy the url somewhere else to preview it

I personally think the previous design dealt with this considerably more effectively (screenshot below), through showing the protocol with `opacity-30`, but still having it present. However, I do understand that the trend for modern browsers is to hide the https:// and www. from user view by default, so perhaps this design should be toggleable, rather than reverted entirely.

<img width="487" src="https://user-images.githubusercontent.com/38821590/165001202-4ddfc945-903f-45b2-8b72-f6702e1e3ddc.png">

**Comment by WorkInProgress-Development** [comment]
This is all related to the new design, which personally I'm not a fan of.
I personally found that the previous design had been the best for me since version 10 (although I miss the clipboard and backspace buttons), and I'd love to see a toggle between the current design (maybe "compact mode") and the v16-17 design.

Version 10 for reference to what I'm talking about
![image](https://user-images.githubusercontent.com/38821590/165002632-e83e7265-1cfd-45ec-b006-ce6c870e4ae8.png) (source: https://github.com/will-stone/browserosaurus/blob/v10.12.2/docs/screenshot.png)

**Reply by will-stone** [reply]
> which personally I'm not a fan of

v17 and all previous versions are available for download if you'd like to stay on an old version. But thanks for all the feedback 👍

**Reply by WorkInProgress-Development** [reply]
Is there any way (that you are aware of) to silence the B icon (in the menubar) turning blue when an updated is required? Possibly some electron parameter?

This is one of the things that has been holding me back from not updating - although when you removed the autoupdate a while back that made a huge difference.

**Reply by will-stone** [reply]
Unfortunately not as that would require back releases which I don't have time to maintain. However, this is the beauty of open source, you can clone and edit things you don't like.
- Clone repo.
- Rollback to commit `4c0a580fe0330d137afce790e8a66142947a4cc7`.
- Comment-out this line: https://github.com/will-stone/browserosaurus/blob/4c0a580fe0330d137afce790e8a66142947a4cc7/src/main/state/middleware.action-hub.ts#L119
- Run `npm i` to install dependencies.
- Run `npm run package:arm` / `npm run package:intel` to obtain the `.app` for your CPU architecture.
- Drag that into your `Applications` to overwrite the one that's there.

**Reply by WorkInProgress-Development** [reply]
I really appreciate you letting me know exactly which line where to comment out - although I'm sure it would be easier for you to just say "do it yourself" (and I could've just found it myself), I appreciate the clear instructions.

**Comment by will-stone** [comment]
The `www.` removal was intended (although `est...` is obviously a bug). I may review this as I didn't expect situations like `www2.`. Thanks.

**Comment by i0ntempest** [comment]
Just coming in to say that I too am not really a fan of the v18 design, especially with v18.1 where the full URL is no longer shown, the backspace function has become a pain to use. I'll open a new thread for discussion on this.
